### ISSUE TITLE
Sound Mode - DOLBY SURROUND not supported

### ISSUE THREAD

**Issue by elad-bar** [ol-iver/denonavr#53]
I have Denon X5200, when playing using sound mode - DOLBY SURROUND - i get an error in HA:
[DenonAVR] Not able to match sound mode, returning raw sound mode.

It also happens when device is turned off

  - platform: denonavr
    host: *******
    name: Denon AVR-X5200W
    show_all_sources: True
    timeout: 2
    sound_mode: True

**Comment by jasonehines** [comment]
I'm also getting this warning.

My iOS Marantz app shows 'Standard(Dolby)' but the details of the card in Home Assistant show nothing selected for Sound Mode. If I select 'Stereo' in the iOS app, it shows 'STEREO' in HA. However if I select 'DOLBY DIGITAL' in HA, it changes the receiver to 'Standard(Dolby)' (could be a limitation of the source from my X1 box) and HA shows nothing in the Sound Mode.

**Comment by starkillerOG** [comment]
@elad-bar and @jasonehines, could you both look at the status of the Receiver in the front panel of HomeAssistant and report the value of the sound_mode_raw propertie when you get this warning?
I have marked the property I mean with the red circle in the picture below:
![sound_mode_raw](https://user-images.githubusercontent.com/14811189/43200287-5eb4aa08-9015-11e8-94d8-a197d99fb446.png)

**Comment by morhimi** [comment]
this is mine:
volume_level: 0.3
is_volume_muted: false
media_content_type: channel
media_title: AUX2
source: AUX2
source_list: AUX2,Blu-ray,Bluetooth,CellcomTV,Chromecast A,Chromecast V,Favorites,Internet Radio,Media Player,Media Server,Online Music,Spotify,TV Audio,Tuner,iPod/USB
sound_mode: DTS Neural:X
sound_mode_list: MUSIC,MOVIE,GAME,AUTO,VIRTUAL,PURE DIRECT,DOLBY DIGITAL,MCH STEREO,STEREO
sound_mode_raw: DTS Neural:X
friendly_name: Denon AVR-X3300W
supported_features: 69004

**Comment by starkillerOG** [comment]
@morhimi, DTS Neural:X has been added to [this PR](https://github.com/scarface-4711/denonavr/pull/52), as soon as that is merged it will be fixed for you.

**Comment by morhimi** [comment]
Thank for the update.
Just so I could understand better, all the supported modes are shown in the sound_mode_list list?

**Comment by starkillerOG** [comment]
@morhimi yes, all modes that HomeAssistant supports are shown in sound_mode_list.
It could be that their are more modes for newer models of receivers, but I did not find anything else in the denonavr documentation, and at least those would then not yet be supported by HomeAssistant.

Note that their are a varity of sound modes like: PLII MUSIC, DTS NEO:6 MUSIC etc that are all mapped to the same MUSIC sound mode.
This is because the receiver will only let you select top level sound modes from the API (in the denon/marantz app) and not specific sound_modes like PLII MUSIC, DTS NEO:6 MUSIC that are available under a sub-menu through the remote of some AVR's

**Comment by jasonehines** [comment]
This is what mine says
sound_mode_raw: Standard(Dolby)

**Comment by starkillerOG** [comment]
@jasonehines I added Standard(Dolby) to [this PR](https://github.com/scarface-4711/denonavr/pull/52), as soon as that is merged it will also be fixed for you.

**Comment by elad-bar** [comment]
that's mine:
sound_mode: Dolby Surround 
sound_mode_list: MUSIC,MOVIE,GAME,AUTO,VIRTUAL,PURE DIRECT,DOLBY DIGITAL,MCH STEREO,STEREO 
sound_mode_raw: Dolby Surround

**Comment by starkillerOG** [comment]
@elad-bar I also added 'Dolby Surround' to [this PR](https://github.com/scarface-4711/denonavr/pull/52), as soon as that is merged it will also be fixed for you.

**Comment by jasonehines** [comment]
this is all mine shows now
`
friendly_name: Living Room Receiver 
supported_features: 3468
`

**Comment by morhimi** [comment]
@jasonehines you need to turn on the receiver first in order to see the relevant information.

**Comment by jasonehines** [comment]
@morhimi The receiver was on. Today with the receiver on I get this
`
volume_level: 0.49 is_volume_muted: false media_content_type: channel media_title: Xfinity source: Xfinity source_list: AUX1,Amazon,AppleTV,BD,GAME,HD Radio,NET/USB,Xfinity friendly_name: Living Room Receiver supported_features: 3468
` 
Also, I don't have the Sound Mode on the card either. I'm not sure why this changes.
![screenshot from 2018-07-27 13 23 11](https://user-images.githubusercontent.com/870809/43337029-82aa7c30-91a0-11e8-9d5b-99c2324e2c78.png)

**Comment by jasonehines** [comment]
I've found another some more on my receiver
`
sound_mode_raw: Pure_Direct sound_mode_raw: Standard(DTS) sound_mode_raw: Neural sound_mode_raw: Multi_CH_Stereo
`
I went through every mode this time, so this should be it for my Marantz.

**Comment by coalfield** [comment]
Some errors on AVR-X4200W spotted following upgrade to .75

```
2018-08-12 12:11:44 WARNING (Thread-7) [DenonAVR] Not able to match sound mode: 'Dolby D + +Dolby Surround', returning raw sound mode.
2018-08-12 15:43:57 WARNING (Thread-17) [DenonAVR] Not able to match sound mode: 'DTS + Neural:X', returning raw sound mode.
2018-08-12 16:18:03 WARNING (Thread-21) [DenonAVR] Not able to match sound mode: 'DTS Surround', returning raw sound mode.

```

**Comment by starkillerOG** [comment]
@jasonehines, @coalfield [This PR](https://github.com/scarface-4711/denonavr/pull/61) will add the sound modes and fix the warnings.

**Comment by ol-iver** [comment]
Available in [0.7.6](https://github.com/scarface-4711/denonavr/releases/tag/0.7.6)
